lib/src/native/realm_core.dart

```diff
@@ -0,0 +1,187 @@
+import 'dart:ffi';
+import 'dart:isolate';
+import 'package:ffi/ffi.dart';
+import 'package:ffi/src/utf8.dart';
+
+import '../configuration.dart';
+import '../realm_class.dart';
+
+import 'realm_bindings.dart';
+
+late RealmLibrary _realmLib;
+
+
+void setRealmLibrary(DynamicLibrary realmLibrary) {
+  _realmLib = RealmLibrary(realmLibrary);
+}
+
+final _RealmCore realmCore = _RealmCore();
+
+class _RealmCore {
+   
+  //From realm.h. Currently not exported from the shared library
+  static const int RLM_INVALID_CLASS_KEY = 0x7FFFFFFF;
+  static const int RLM_INVALID_PROPERTY_KEY = -1;
+  static const int RLM_INVALID_OBJECT_KEY = -1;
+  static const int SCHEDULER_FINALIZE = -1;
+
+  // Hide the RealmCore class and make it a singleton
+  static _RealmCore? _instance;
+
+  _RealmCore._();
+
+  factory _RealmCore() {
+    return _instance ??= _RealmCore._();
+  }
+  //
+
+  String get libraryVersion => _realmLib.realm_get_library_version().cast<Utf8>().toDartString();
+
+  //TODO: Use Finalizers, when available, instead of native WeakHandles https://github.com/dart-lang/language/issues/1847
```

Comment:
Would loved this to be a code suggestion :) I will mention it sure, but imo the issue link should be enough most of the time
```suggestion
  //TODO: Use Finalizers, when available, instead of native WeakHandles https://github.com/dart-lang/language/issues/1847
  //The proposal: https://github.com/dart-lang/language/blob/master/working/1847%20-%20FinalizationRegistry/proposal.md
```